File: src/types/patient.ts
// src/types/patient.ts

export interface PatientHistoryEntry {
  date: string;
  type: 'Consultation' | 'Treatment' | 'Follow-up' | 'Lab Test' | 'Therapy Session';
  notes: string;
}

export interface PatientAvailability {
  start: string;
  end: string;
}

export interface Patient {
  id: string;
  name: string;
  age: number;
  gender: 'Male' | 'Female' | 'Other';
  email: string;
  phone: string;
  complaint: string;
  conditions: string[];
  history: PatientHistoryEntry[];
  availability: PatientAvailability[];
}

export interface PatientState {
  patients: Patient[];
  selectedPatient: Patient | null;
  isLoading: boolean;
  error: string | null;
}

File: server/app/models/patient.py
from typing import List, Literal
from beanie import Document
from pydantic import BaseModel


class PatientHistoryEntry(BaseModel):
    date: str
    type: Literal["Consultation", "Treatment", "Follow-up", "Lab Test", "Therapy Session"]
    notes: str


class PatientAvailability(BaseModel):
    start: str
    end: str


class Patient(Document):
    pid: str
    name: str
    age: int
    gender: Literal["Male", "Female", "Other"]
    email: str
    phone: str
    complaint: str
    conditions: List[str] = []
    history: List[PatientHistoryEntry] = []
    availability: List[PatientAvailability] = []

    class Settings:
        name = "patients"
